sdk/src/types/block/protocol.rs

```diff
@@ -20,27 +27,82 @@ use crate::types::block::{helper::network_name_to_id, output::RentStructure, Con
+    // The work score structure used by the node/network.
+    #[getset(get = "pub")]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    work_score_structure: Option<WorkScoreStructure>,
     // TokenSupply defines the current token supply on the network.
-    #[cfg_attr(feature = "serde", serde(with = "crate::utils::serde::string"))]
+    #[serde(with = "crate::utils::serde::string")]
+    #[getset(get_copy = "pub")]
     token_supply: u64,
     // Genesis timestamp at which the slots start to count.
-    #[cfg_attr(feature = "serde", serde(alias = "genesisUnixTimestamp"))]
+    #[serde(with = "crate::utils::serde::string")]
+    #[getset(get_copy = "pub")]
     genesis_unix_timestamp: u32,
     // Duration of each slot in seconds.
-    #[cfg_attr(feature = "serde", serde(alias = "slotDurationInSeconds"))]
+    #[getset(get_copy = "pub")]
     slot_duration_in_seconds: u8,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    slots_per_epoch_exponent: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    mana_generation_rate: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    mana_generation_rate_exponent: Option<u32>,
+    #[packable(unpack_error_with = |err| Error::InvalidManaDecayFactors(err.into_opt_error()))]
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(skip)]
+    mana_decay_factors: Option<BoxedSlicePrefix<u32, u8>>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    mana_decay_factors_exponent: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    mana_decay_factor_epochs_sum: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    mana_decay_factor_epochs_sum_exponent: Option<u32>,
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "crate::utils::serde::option_string"
+    )]
+    #[getset(get_copy = "pub")]
+    staking_unbonding_period: Option<u32>,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    #[getset(get_copy = "pub")]
+    eviction_age: Option<SlotIndex>,
+    // TODO: wtf are these? should they be strings?
+    #[packable(unpack_error_with = |err| Error::InvalidLivenessThreshold(err.into_opt_error()))]
+    #[serde(
+        default,
+        skip_serializing_if = "Option::is_none",
+        with = "crate::utils::serde::option_string_prefix"
+    )]
+    #[getset(skip)]
+    liveness_threshold: Option<StringPrefix<u8>>,
+    #[packable(unpack_error_with = |err| Error::InvalidEpochNearingThreshold(err.into_item_err()))]
+    #[serde(with = "crate::utils::serde::string_prefix")]
+    #[getset(skip)]
+    epoch_nearing_threshold: StringPrefix<u8>,
```

Comment:
I think this should be `SlotIndex`

Reply:
agreed, tip example is just a number as well